### PR TITLE
Sidebar redesign

### PR DESCRIPTION
**Pull request by ewels**
Making a start on redesigning the docs nav to be entirely in the sidebar.

Todo:

- [x] Make the product picker interactive (currently doesn't do anything)
- [x] Tidy up light / dark mode switcher styles at the bottom
- [x] Check / fix mobile styling
- [x] Add platform version picker
- [x] Add sidebar nav to docs homepage with product buttons

### CODE REVIEW

**Comment by MythicalFish** [comment]
Great start!

**Comment by deepcshah** [comment]
Screenshot 2024-09-23 at 5 33 32 PM

omg so nice

Nits for version selection: 
1. Should the arrow for the version selector also point right when not in use and animate just like the product dropdown?
2. The text color for the versions are a bit too gray, I think we can use a darker shade

@ewels

**Comment by ewels** [comment]
@deepcshah - thanks for the feedback! I was still building it when you wrote that actually 😅 Both things should be resolved now. Let me know if you spot anything else.

Approaching complete now. Need to fix a bunch of broken stuff on mobile, and ideally clean up the code a bit, but I think that's it. Hopefully the difficult stuff is done now.

**Review by MythicalFish — APPROVED** [review]
Can't seem to repro the faulty switcher, but apart from that looks good! Good enough to merge I think

**Comment by ewels** [comment]
ok not sure what changed, but I can't reproduce (locally or in preview) it now either, so presumably fixed! 🎉 

@deepcshah defer to you on the design changes, but from my side LGTM to merge 👍🏻

File: src/theme/DocSidebar/Desktop/ProductSwitcher/styles.module.css [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    }
+  }
+}
+
+.item {
+  height: 34px;
+  display: flex;
+  align-items: start;
+  justify-content: space-between;
+  padding: 10px 12px;
```
Comment (ewels) [inline src/theme/DocSidebar/Desktop/ProductSwitcher/styles.module.css:27]:
```suggestion
  padding: 0 12px;
```

File: src/theme/DocSidebar/Desktop/ProductSwitcher/styles.module.css [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+      transition: transform var(--ifm-transition-fast) linear;
+      transform: rotate(-90deg);
+    }
+  }
+}
+
+.item {
+  height: 34px;
+  display: flex;
+  align-items: start;
```
Comment (ewels) [inline src/theme/DocSidebar/Desktop/ProductSwitcher/styles.module.css:25]:
```suggestion
  align-items: center;
```

File: src/theme/DocSidebar/Desktop/ProductSwitcher/ProductLogo/index.jsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    Fusion: {
+      light: Fusion,
+      dark: FusionDark,
+    },
+  };
+  const Logo = logos[product].light;
+  const LogoDark = logos[product].dark;
+  return (
+    <span
+      className={clsx(styles.item, { [styles.lg]: product === "Platform" })}
```
Comment (ewels) [inline src/theme/DocSidebar/Desktop/ProductSwitcher/ProductLogo/index.jsx:63]:
```suggestion
      className={styles.item}
```

File: src/theme/DocSidebar/Desktop/ProductSwitcher/ProductLogo/styles.module.css [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    width: 100%;
+    height: 14px;
+  }
+  &.lg {
+    & svg {
+      height: 16px;
+      position: relative;
+      top: -1px;
+    }
+  }
```
Comment (ewels) [inline src/theme/DocSidebar/Desktop/ProductSwitcher/ProductLogo/styles.module.css:21]:
```suggestion
  & svg {
    width: 100%;
    height: 15px;
  }
```

**Review by ewels — COMMENTED** [review]
Ok thought I was going completely crazy thinking that the platform logo was _slightly_ bigger than the others. But I wasn't, it's hardcoded as such.

Removing this and making them all meet in the middle at 15px stops my eyebrow from flinching every time I open the dropdown.

**Pull request merged**